specs/experimental/alligator.md

```diff
@@ -0,0 +1,401 @@
+    uint256 allowance;
+}
+```
+
+| Name                     | Type            | Description                                                             |
+|--------------------------|-----------------|-------------------------------------------------------------------------|
+| `maxRedelegations`       | `uint256`       | Maximum number of times the delegated votes can be redelegated.         |
+| `notValidBefore`         | `uint256`       | Timestamp after which the delegation is valid.                          |
+| `notValidAfter`          | `uint256`       | Timestamp before which the delegation is valid.                         |
+| `allowanceType`          | `AllowanceType` | Type of allowance (e.g., absolute or relative).                         |
+| `allowance`              | `uint256`       | Amount of votes delegated, denominated in the token's decimals.         |
+
+### `AllowanceType`
+
+Subdelegations can have different types of allowances, represented with:
+
+```solidity
+enum AllowanceType {
+  Absolute,
+  Relative
+}
+```
+
+| Name        | Number | Description                                                                                 |
+|-------------|--------|---------------------------------------------------------------------------------------------|
+| `Absolute`  | `0`    | The amount of votes delegated is fixed.                                                     |
+| `Relative`  | `1`    | The amount of votes delegated is relative to the total amount of votes the delegator has.   |
+
+## Backwards Compatibility
+
+The `Alligator` contract ensures backwards compatibility by allowing the migration of delegation state from the
+token contract.
+
+## User Flow
+
+The following sections highlight the use cases that MUST be supported by subdelegations, and the difference for basic
+delegations made from the `GovernanceToken` contract.
+
+### Partial delegations
+
+Users MUST be able to perform partial delegations of their voting power to another address. Additionally, the `Alligator`'s
+subdelegation rules MUST allow users to perform subdelegations for relative and absolute amounts of voting power. Absolute
+amounts MUST be denominated in the token's decimals, whereas relative amounts MUST be denominated in percentages.
+
+### Constrained delegations
+
+Users MUST be able to perform time & block constrained delegations. The subdelegation rules MUST include optional
+`notValidBefore` and `notValidAfter` fields to allow users to specify the time range in which a delegation is valid, both
+denominated as timestamps.
+
+### Redelegations
+
+Users MUST be able to redelegate their voting power to another address. If a user delegates their voting power to another
+address, this second address can further delegate the voting power to a third address. The first delegator MUST be able to
+limit the number of redelegations that can be performed for their delegated voting power by setting the `maxRedelegations`
```

Comment:
Is the convention to set this value to `type(maxRedelgations).max` to represent "infinite redelegations"?

Reply:
removed redelegations as it was out of scope